### PR TITLE
Permission Load Fix

### PR DESCRIPTION
**Pull request by overkidding**
When i created the Checks Command PR i actually encountered an error and when i joined the server the "exempted" variable in the check was never true and i fixed it by checking every pollData the permissions for all the checks.

### CODE REVIEW

**Comment by SamB440** [comment]
Can you fix the spacing formatting?

File: src/main/java/ac/grim/grimac/player/GrimPlayer.java [hunk 1]
```diff
@@ -504,6 +504,8 @@ public void updatePermissions() {
         if (bukkitPlayer == null) return;
         this.noModifyPacketPermission = bukkitPlayer.hasPermission("grim.nomodifypacket");
         this.noSetbackPermission = bukkitPlayer.hasPermission("grim.nosetback");
+        FoliaScheduler.getAsyncScheduler().runNow(GrimAPI.INSTANCE.getPlugin(), t-> checkManager.allChecks.values().stream().map(c->(Check)c)
+                .forEach(Check::updateExempted));
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/player/GrimPlayer.java]: Do not cast without checking
```suggestion
        FoliaScheduler.getAsyncScheduler().runNow(GrimAPI.INSTANCE.getPlugin(), t ->
            for (AbstractCheck check : checkManager.allChecks.values()) {
                if (check instanceof Check) {
                    ((Check) check).updateExempted();
                }
            }
        );
```

File: src/main/java/ac/grim/grimac/checks/Check.java [hunk 2]
```diff
@@ -60,6 +61,13 @@ public boolean shouldModifyPackets() {
         return isEnabled && !player.disableGrim && !player.noModifyPacketPermission && !exempted;
     }
 
+    public void updateExempted(){
+        if(player.bukkitPlayer == null || checkName == null) return;
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/checks/Check.java]:
```suggestion
        if (player.bukkitPlayer == null || checkName == null) return;
```

File: src/main/java/ac/grim/grimac/checks/Check.java [hunk 3]
```diff
@@ -60,6 +61,13 @@ public boolean shouldModifyPackets() {
...
+        FoliaScheduler.getEntityScheduler().run(player.bukkitPlayer, GrimAPI.INSTANCE.getPlugin(),
+                t -> exempted = player.bukkitPlayer != null && player.bukkitPlayer.hasPermission("grim.exempt." + checkName.toLowerCase()),
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/checks/Check.java]: You already checked if bukkitPlayer was null
```suggestion
                t -> exempted = player.bukkitPlayer.hasPermission("grim.exempt." + checkName.toLowerCase()),
```

File: src/main/java/ac/grim/grimac/checks/Check.java [hunk 4]
```diff
@@ -60,6 +61,13 @@ public boolean shouldModifyPackets() {
...
+                ()->{});
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/checks/Check.java]:
```suggestion
                () -> {});
```

**Review by ManInMyVan — CHANGES_REQUESTED** [review]

File: src/main/java/ac/grim/grimac/checks/Check.java [hunk 5]
```diff
@@ -60,6 +61,13 @@ public boolean shouldModifyPackets() {
         return isEnabled && !player.disableGrim && !player.noModifyPacketPermission && !exempted;
     }
 
+    public void updateExempted(){
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/checks/Check.java]:
```suggestion
    public void updateExempted () {
```

**Review by ManInMyVan — CHANGES_REQUESTED** [review]

**Comment by ManInMyVan** [comment]
I think we should also remove spaces from check names, otherwise we get things like `grim.exempt.timer - vehicle`, which can't be be given with luckperms in-game.

File: src/main/java/ac/grim/grimac/checks/Check.java [hunk 6]
```diff
@@ -60,6 +61,13 @@ public boolean shouldModifyPackets() {
         return isEnabled && !player.disableGrim && !player.noModifyPacketPermission && !exempted;
     }
 
+    public void updateExempted () {
```
Comment (ManInMyVan) [inline src/main/java/ac/grim/grimac/checks/Check.java]: I added a space here for some reason, oops
```suggestion
    public void updateExempted() {
```

**Review by ManInMyVan — CHANGES_REQUESTED** [review]

**Review by ManInMyVan — APPROVED** [review]

**Pull request merged**